cmd/snap/main.go

```diff
@@ -130,11 +130,16 @@ func init() {
 	}
 }
 
+var missingLoginHelp = "Access denied (snap login --help).\n"
+
 func main() {
 	if err := run(); err != nil {
 		if e, ok := err.(*flags.Error); ok && e.Type == flags.ErrHelp {
 			fmt.Fprintf(Stdout, "%v\n", err)
 			os.Exit(0)
+		} else if e, ok := err.(*client.Error); ok && e.StatusCode == 401 && e.Message == "Unauthorized" {
```

Comment:
This is still comparing the error message. I suggest making this decision based on whether the status is 401 and we have authorization data at hand (client.LoggedIn?). If we are logged in, pointing out the help won't help much. If we aren't logged in, a 401 is asking us to do so.